docs(nms): Add contributing conventions for JavaScript (#8174)
magma/magma: docs/readmes/contributing/contribute_conventions.md

- Unit tests should default to being placed in the same directory as the files they're testing, except for the following
    - Python: directly-adjacent `tests` directory
    - C/C++: directly-adjacent `tests` directory

>     - Javascript: directly-adjacent `__tests__` directory

Reviewer: ```suggestion
    - JavaScript: directly-adjacent `__tests__` directory
```

    - Default to `.bash` except with specific reason
- When a shell script passes around 100 lines, it's time to re-write it in Python or Go

### Javascript

**General**

> - Importing order should be: types, default components, and named components with each separated by a newline

Reviewer: ```suggestion
- Import order should be: types, default components, and named components with each separated by a newline
```

> - Use `CamelCase` for component file names

Reviewer 2: Should be TitleCase

- Favour `map` and `forEach` functions in place of regular `for` loops
- Refrain from using literal strings/numbers without defining them
- Use strict equality(`===`) when comparing values

> **Type Annotations**

Reviewer 3: I think for **Type Annotations** and **Linter** we might want to add a line to link the dev_testing instructions.

Author: Yeah, that was what I think of too. So, I linked flow and eslint official website for now but will change them to link to dev_testing instruction when that pull request gets approved cause that makes more sense.

Reviewer: ```suggestion
**Type annotations**
```

Author: @themarwhal I have now added the link to the dev_testing instructions since that got merged.

- All new code should be fully type anotatated

> - For mandatory type checks, we have a mandatory unit test that runs [flow](https://flow.org/) on the nms codebase

Reviewer: ```suggestion
- We have a mandatory unit test that runs the [flow](https://flow.org/) type checker on the NMS codebase
```

    - On CI, the check gets run as part of the `nms-flow-test` job

**Documentation**

> - Document all the components and functions you created

Reviewer: ```suggestion
- Document all the components and functions you create
```

- Keep the docs up to date when you make changes
- Use [JSDoc](https://jsdoc.app/index.html) tags to document your code
- Use `//` to comment line of code, if it is not clear in what it is doing

**Linter**

> - Run [eslint](https://eslint.org/) to lint your changes locally

Reviewer: ```suggestion
- Run [ESLint](https://eslint.org/) to lint your changes locally
```
